hr_public_holidays/wizards/public_holidays_next_year_wizard.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Trobz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, models, fields
+
+
+_logger = logging.getLogger(__name__)
+
+
+class PublicHolidaysNextYearWizard(models.TransientModel):
+    _name = 'public.holidays.next.year.wizard'
+    _description = 'Creates public holidays for the year following the last'
+
+    @api.multi
+    def create_next_year_public_holidays(self):
+
+        ph_env = self.env['hr.holidays.public']
+        pholidays = ph_env.search([])
+
+        NO_COUNTRY = 'NO_COUNTRY'
+        last_ph_dict = {}
+        for ph in pholidays:
+
+            curr_country = ph.country_id and ph.country_id.id or NO_COUNTRY
```

Comment:
idiomatic python: `curr_country = ph.country_id.id if ph.country_id else NO_COUNTRY`
But you could even put directly the country recordset as the dict keys  and skip the check altoghether. This is possible because [recordset implements the \_\_hash\_\_ method](https://github.com/odoo/odoo/blob/10.0/odoo/models.py#L5157).
`last_ph_country = last_ph_dict.get(ph.country_id, False)`